base/windows/djinn/DjinnListWindow.ts

```diff
@@ -67,6 +67,7 @@ export class DjinnListWindow {
     public selected_djinn_index: number;
     public action_text_selected: boolean;
     public page_index: number;
+    public djinn_chosen: boolean;
```

Comment:
Let's see whether those `this.grant_full_user_control()` calls are necessary. Then we can decide whether this variable is truly necessary.